custom_model_runner/datarobot_drum/drum/root_predictors/utils.py

```diff
@@ -102,6 +104,41 @@ def _create_custom_model_dir(
     return custom_model_dir
 
 
+def _queue_output(stdout, stderr, queue):
+    """Helper function to stream output from subprocess to a queue."""
+    for line in iter(stdout.readline, b""):
+        queue.put(line)
+    for line in iter(stderr.readline, b""):
+        queue.put(line)
+    stdout.close()
+    stderr.close()
+
+
+def _stream_p_open(subprocess_popen: subprocess.Popen):
+    """Wraps the Popen object to stream output in a separate thread.
+    This realtime output of the stdout and stderr of the process
+    is streamed to the terminal.
+    """
+    q = Queue()
+    t = Thread(target=_queue_output, args=(subprocess_popen.stdout, subprocess_popen.stderr, q))
+    t.daemon = True  # thread dies with the program
+    t.start()
+    while True:
+        try:
+            # Stream output if available
+            line = q.get_nowait()
+            logger.info(line.strip()) if len(line.strip()) > 0 else None
```

Comment:
Do I understand correct that it only shows logs to tell user that process is alive. 
But the actual output of  popen(drum .. .. )  is used

Reply:
Note.
THere is also a `StdoutFlusher` thread used to flush :)
Maybe you could simply add logging into it?